woltka/workflow.py

```diff
@@ -0,0 +1,590 @@
+        Whether perform demultiplexing.
+    """
+    # read sample Ids
+    if ids:
+        samples = read_ids(ids)
+        click.echo(f'Samples to include: {len(samples)}.')
+    else:
+        samples = None
+
+    errmsg = 'Provided sample IDs and actual files are inconsistent.'
+
+    # path is an alignment file
+    if isfile(path_):
+
+        # turn on demultiplexing if not decided
+        demux = demux is not False
+        if demux:
+            files = [path_]
+
+        # validate with given sample Ids
+        else:
+            sample = path2stem(path_, ext)
+            if samples and samples != [sample]:
+                raise ValueError(errmsg)
+            files = {path_: sample}
+            samples = [sample]
+
+    # path is a directory of alignment files
+    elif isdir(path_):
+
+        # turn off demultiplexing if not decided
+        demux = demux or False
+
+        # get a map of plausible sample Ids to files
+        map_ = id2file_map(path_, ext, not demux and samples)
+        if len(map_) == 0:
+            raise ValueError('No valid file found in directory.')
+        if demux:
+            files = sorted([join(path_, x) for x in map_.values()])
+
+        # validate with given sample Ids
+        else:
+            if not samples:
+                samples = sorted(map_.keys())
+            elif len(map_) < len(samples):
+                raise ValueError(errmsg)
+            files = {join(path_, map_[x]): x for x in samples}
+
+    else:
+        raise ValueError(f'"{path_}" is not a valid file or directory.')
+
+    click.echo(f'Alignment files to read: {len(files)}.')
+    click.echo(f'Demultiplexing: {"on" if demux else "off"}.')
+    return samples, files, demux
+
+
+def build_align_proc(coords_fp: str = None,
+                     overlap:   int = None) -> object:
+    """Build alignment processor.
+
+    Parameters
+    ----------
+    coords_fp : str, optional
+        Path to gene coordinates file.
+    overlap : int, optional
+        Read/gene overlapping percentage threshold
+
+    Returns
+    -------
+    object
+        Alignment processor.
+
+    Notes
+    -----
+    Currently two processors are supported: Plain() for regular alignments
+    (i.e., simple query-to-subject maps), Ordinal() for alignments with
+    coordinates which will be used to match queries (reads) and genes. The
+    presence of a gene coordinates file (`coords_fp`) is an indicator for
+    using the latter.
+    """
+    if coords_fp:
+        click.echo(f'Reading gene coordinates...', nl=False)
+        with readzip(coords_fp) as fh:
+            coords = read_gene_coords(fh, sort=True)
+        click.echo(' Done.')
+        click.echo(f'Total host sequences: {len(coords)}.')
+        return Ordinal(coords, whether_prefix(coords),
+                       overlap and overlap / 100)
+    else:
+        return Plain()
+
+
+def build_hierarchy(names_fp:   str = None,
```

Comment:
It appears like this function has a number of acceptable tree formats, and if multiple trees are specified,
e.g.,
```
build_hierarchy(
    nodes_fp='nodes.txt', newick_fp='tree.nwk',
)
```
and as long as `nodes.txt` and `tree.nwk` do not have any shared nodes with differing parents, then edges from both trees will be used.

I think this behavior should either be more explicitly documented somewhere, or expressly disallowed, depending on your intended use.

Reply:
I added bit more information to `README.md` to explain how it works.

Reply:
Alright that works.